Manifold/Expression.swift

```diff
@@ -4,8 +4,9 @@ public enum Expression<Recur> {
 	case Type(Int)
 	case Variable(Name)
 	case Application(Recur, Recur)
-	case Lambda(Int, Recur?, Recur)
+	case Lambda(Int, Recur, Recur)
```

Comment:
The new `Implicit` expression can be used for lambda type annotations in place of optionals.